example_dart/lib/queued_interceptor_crsftoken.dart

```diff
@@ -1,164 +1,171 @@
-    '/response-headers',
+  final tokenManager = TokenManager();
 
-    /// Pretend the Server has generated CSRF token
-    /// and passed it to the client.
-    queryParameters: {
-      headerKey: 'fbf07f2b-b957-4555-88a2-3d3e30e5fa64',
-    },
+  final dio = Dio(
+    BaseOptions(
+      baseUrl: 'https://httpbun.com/',
+    ),
   );
-  printLog(1, accessResult.realUri.path);
-
-  /// #2 Make a request(POST) to the Server
-  ///
-  /// Pretend the token has expired.
-  ///
-  /// Then the interceptor will request a new CSRF token
-  final createResult = await dio.post(
-    '/mix/s=401/',
+
+  dio.interceptors.add(
+    QueuedInterceptorsWrapper(
+      onRequest: (requestOptions, handler) {
+        print(
+          '''
+[onRequest] ${requestOptions.hashCode} / time: ${DateTime.now().toIso8601String()}
+\tPath: ${requestOptions.path}
+\tHeaders: ${requestOptions.headers}
+          ''',
+        );
+
+        /// In case, you have 'refresh_token' and needs to refresh your 'access_token'
+        /// Make a request for new 'access_token' and update from here
+
+        if (tokenManager.accessToken != null) {
+          requestOptions.headers['Authorization'] =
+              'Bearer ${tokenManager.accessToken}';
+        }
+
+        return handler.next(requestOptions);
+      },
+      onResponse: (response, handler) {
+        print('''
+[onResponse] ${response.requestOptions.hashCode} / time: ${DateTime.now().toIso8601String()}
+\tStatus: ${response.statusCode}
+\tData: ${response.data}
+        ''');
+
+        return handler.resolve(response);
+      },
+      onError: (error, handler) async {
+        final statusCode = error.response?.statusCode;
+        print(
+          '''
+[onError] ${error.requestOptions.hashCode} / time: ${DateTime.now().toIso8601String()}
+\tStatus: $statusCode
+          ''',
+        );
+
+        /// This example only handles '401' status code,
+        /// The more complex scenario should handle more status codes e.g. '403', '404', etc.
+        if (statusCode != 401) {
+          return handler.resolve(error.response!);
+        }
+
+        /// Consider [dio] can be requested in parallel.
+        ///
+        /// To prevent repeated requests to the 'Authentication Server'
+        /// to update our 'access_token',
+        /// we can compare with the previously requested 'access_token'.
+        final requestedAccessToken =
+            error.requestOptions.headers['Authorization'];
+        if (requestedAccessToken == tokenManager.accessToken) {
+          final tokenRefreshDio = Dio()
+            ..options.baseUrl = 'https://httpbun.com/';
+
+          final response = await tokenRefreshDio.post(
+            'https://httpbun.com/mix/s=201/b64=${base64.encode(
+              jsonEncode(AuthenticationServer.generate()).codeUnits,
+            )}',
+          );
+          if (response.statusCode == null || response.statusCode! ~/ 100 != 2) {
+            return handler.reject(error);
```

Comment:
tokenRefreshDio need close

Reply:
I added closing lines for `tokenRefreshDio`